### PR TITLE
drivers: i2c: enable i2c target mode for LPI2C driver

### PR DESCRIPTION
**Pull request by danieldegrasse**
Enable I2C target mode for LPI2C driver. Verified with i2c_target_api
test, on RT1060 EVK.

### CODE REVIEW

File: drivers/i2c/i2c_mcux_lpi2c.c [hunk 1]
```diff
@@ -297,12 +305,175 @@ static int mcux_lpi2c_recover_bus(const struct device *dev)
...
+					(kLPI2C_SlaveTxReadyFlag |
+					kLPI2C_SlaveRxReadyFlag |
+					kLPI2C_SlaveStopDetectFlag |
+					kLPI2C_SlaveAddressValidFlag |
+					kLPI2C_SlaveTransmitAckFlag));
+	return 0;
+}
+
+static int mcux_lpi2c_target_unregister(const struct device *dev,
+			       struct i2c_target_config *target_config)
```
Comment (bbilas) [inline drivers/i2c/i2c_mcux_lpi2c.c:447]: Indentation.

File: drivers/i2c/i2c_mcux_lpi2c.c [hunk 2]
```diff
@@ -297,12 +305,175 @@ static int mcux_lpi2c_recover_bus(const struct device *dev)
...
+			target_cb->stop(data->target_cfg);
+		}
+	}
+	if (flags & kLPI2C_SlaveTransmitAckFlag) {
+		LPI2C_SlaveTransmitAck(config->base, data->send_ack);
+	}
+}
+
+static int mcux_lpi2c_target_register(const struct device *dev,
+			struct i2c_target_config *target_config)
```
Comment (bbilas) [inline drivers/i2c/i2c_mcux_lpi2c.c:399]: Indentation.

**Comment by bbilas** [comment]
I've done basic tests on my custom board with RT1062 MCU and it works well.

File: drivers/i2c/i2c_mcux_lpi2c.c (hunk 1 above)
Comment (danieldegrasse) [reply drivers/i2c/i2c_mcux_lpi2c.c:447]: I have no problem adjusting the indentation here, but what are you looking for? Alignment with the prior `(`? I don't see any specific guidelines on how to indent the second line of a function declaration within Zephyr.

**Review by teburd — DISMISSED** [review]

**Comment by danieldegrasse** [comment]
@bbilas formatting suggestions should be addressed

**Review by teburd — APPROVED** [review]

**Review by bbilas — APPROVED** [review]

**Pull request merged**